docs/features/image_name_substitution.md

```diff
@@ -77,6 +77,13 @@ Testcontainers will not apply the prefix to:
 * non-Hub image names (e.g. where another registry is set)
 * Docker Hub image names where the hub registry is explicitly part of the name (i.e. anything with a `docker.io` or `registry.hub.docker.com` host part)
 
+If you want your registry to handle both official Docker Hub images (e.g `postgres`)
+as well as images from other registries (e.g `mycompany/postgres`), you can use the
+`TESTCONTAINERS_HUB_IMAGE_NAME_NORMALIZE` environment variable or the `hub.image.name.normalize`
+configuration option. When set to `true`, Testcontainers will normalize the official Docker Hub
+image names to start with `library/`. When this option is used, Testcontainers will additionally
+disable image compatibility checks done by some containers, so it the compatibility responsibility
+is on the user.
```

Comment:
Added this. cc @eddumelendez.

Did i get the env variable name right?
There is automatic conversion from TESTCONTAINERS_HUB_IMAGE_NAME_NORMALIZE to hub.image.name.normalize, right?

Reply:
yes, the name is correct.

> When this option is used, Testcontainers will additionally disable image compatibility checks done by some containers, so it the compatibility responsibility is on the user.

I don't think this is the case because the check compatibility is executed first. Can you please add a test similar to https://github.com/testcontainers/testcontainers-java/blob/00cc0ea84b73642e3494e6b7a0384a5e3fa201bd/core/src/test/java/org/testcontainers/utility/ImageNameSubstitutorTest.java#L58-L68 in order to cover it?

Reply:
> I don't think this is the case because the check compatibility is executed first.

that makes sense!

however, when testing this PR internally, i run into some issues until i added "as compatible substitute" call.
is it possible that there are some additional checks somewhere else? sadly, don't have stacktrace now